pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -180,6 +181,24 @@ private void initInstanceDataDir(File instanceDataDir) {
     }
   }
 
+  @Override
+  public List<File> getConsumerDirPaths() {
+    String consumerDirPath = _instanceDataManagerConfig.getConsumerDir();
+
+    if (consumerDirPath != null) {
+      return Collections.singletonList(new File(consumerDirPath));
+    } else {
+      List<File> consumerDirs = new ArrayList<>();
+      String instanceDataDir = _instanceDataManagerConfig.getInstanceDataDir();
+      for (String tableNameWithType : getAllTables()) {
+        String dirPath = instanceDataDir + File.separator + tableNameWithType + File.separator
+            + RealtimeTableDataManager.CONSUMERS_DIR;
+        consumerDirs.add(new File(dirPath));
+      }
+      return consumerDirs;
```

Comment:
My understanding is that if the consumerDir is not specified, the table is not mmapping the consuming segments (it uses direct offheap). So maybe we don't need the else part?!

Reply:
When consumerDir is not specified, `RealtimeTableDataManager` constructs a consumer directory under table data directory (`<path-to-dataDir>/<table_name>/consumers`). Else case here is to cover that case.